File: 435.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        int ans = 0;
        int n = intervals.size();
        int i = 0;
        sort(intervals.begin(), intervals.end());
        while(i < n-1){
            if(intervals[i][1] > intervals[i+1][0]){
                if(intervals[i][1] < intervals[i+1][1]){
                    intervals[i+1] = intervals[i];
                }
                ans++;
            }
            i++;
        }
        return ans;
    }
};

File: Python/435.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort()
        end = intervals[0][1]
        sol = 0
        if len(intervals) <= 1:
            return 0
        i = 1
        print(intervals)
        while i < len(intervals):
            if end > intervals[i][0]:
                if end > intervals[i][1]:
                    end = intervals[i][1]
                    sol += 1
                else:
                    sol+=1
            else:
                end = intervals[i][1]
            i+= 1
        return sol
